.github/actions/specs-report/__tests__/main.test.ts

```diff
@@ -0,0 +1,172 @@
+
+// Mock action used functions
+let getFilesMock: jest.SpiedFunction<typeof files.getFiles>
+let readJsonFileMock: jest.SpiedFunction<typeof files.readJsonFile>
+let junitParseMock: jest.SpiedFunction<typeof junit2json.parse>
+let addCommentToPrMock: jest.SpiedFunction<typeof prComment.addCommentToPr>
+
+const defaultGetInputMockImplementation = (): void => {
+  getInputMock.mockImplementation(name => {
+    switch (name) {
+      case 'junit-report-paths':
+        return '500'
+      case 'spec-path':
+        return '500'
+      case 'git-token':
+        return 'fake-token'
+      case 'comment-on-pr':
+        return 'true'
+      case 'fail-on-missing-vectors':
+        return 'false'
+      case 'fail-on-failed-test-cases':
+        return 'true'
+      default:
+        return ''
+    }
+  })
+}
+
+const VALID_SPEC_JSON_BODY = {
+  description: 'This is a valid tbdex spec',
+  input: {},
+  output: {}
+}
+
+describe('action', () => {
+  let output: Record<string, string> = {}
+
+  beforeEach(() => {
+    jest.clearAllMocks()
+    output = {}
+
+    getInputMock = jest.spyOn(core, 'getInput').mockImplementation()
+    setOutputMock = jest
+      .spyOn(core, 'setOutput')
+      .mockImplementation((k, v) => (output[k] = v))
+    setFailedMock = jest.spyOn(core, 'setFailed').mockImplementation()
+    errorMock = jest.spyOn(core, 'error').mockImplementation()
+    summaryWriteMock = jest.spyOn(core.summary, 'write').mockImplementation()
+
+    getFilesMock = jest.spyOn(files, 'getFiles').mockImplementation()
+    readJsonFileMock = jest.spyOn(files, 'readJsonFile').mockImplementation()
+
+    addCommentToPrMock = jest
+      .spyOn(prComment, 'addCommentToPr')
+      .mockImplementation()
+
+    junitParseMock = jest.spyOn(junit2json, 'parse')
+  })
+
+  it('generates a report without any failures', async () => {
+    defaultGetInputMockImplementation()
+    getFilesMock
+      // get junit report xml first
+      .mockReturnValueOnce(Promise.resolve(SUCCESS_MOCK.junitFiles))
+      // get test vector json
+      .mockReturnValueOnce(Promise.resolve(TBDEX_TEST_VECTORS_FILES))
+    readJsonFileMock.mockReturnValue(VALID_SPEC_JSON_BODY)
+
+    const rawJunitFile = files.readFile(SUCCESS_MOCK.junitFiles[0])
+
+    await main.run()
+    expect(runMock).toHaveReturned()
+
+    expect(junitParseMock).toHaveReturnedTimes(1)
+    expect(junitParseMock).toHaveBeenCalledWith(rawJunitFile)
+
+    console.info({ output })
+    expect(setOutputMock).toHaveBeenCalled()
+    expect(output.success).toBe('true')
+    expect(setFailedMock).not.toHaveBeenCalled()
+    expect(errorMock).not.toHaveBeenCalled()
+
+    expect(output.summary).toMatch('✅ All test vectors passed')
+    expect(summaryWriteMock).toHaveBeenCalledTimes(1)
+
+    expect(addCommentToPrMock).toHaveBeenCalledWith(
+      output.summary,
+      'fake-token'
+    )
+
+    const expectedReport = {
+      totalJunitFiles: 1,
+      totalTestVectors: 10,
+      totalJunitTestCases: 99,
```

Comment:
will we have to keep updating this?

Reply:
nope because its based on the mocked xml result (see the `__tests__/mocks.ts`)